ion-schema/src/main/kotlin/com/amazon/ionschema/model/TypeArgument.kt

```diff
@@ -0,0 +1,46 @@
+package com.amazon.ionschema.model
+
+/**
+ * A TypeArgument represents (defines or references) a Type, allowing the Type to be used as an argument for a constraint.
+ *
+ * **_Do not implement this interface_**—It will become a sealed interface once `ion-schema-kotlin` is updated to use
+ * language version 1.5.
+ */
+@ExperimentalIonSchemaModel
+sealed class TypeArgument {
+    abstract val nullability: Nullability
+
+    /**
+     * Nullability modifiers for [TypeArgument].
+     */
+    enum class Nullability {
+        /**
+         * No special treatment of null values.
+         */
+        None,
+        /**
+         * Ion Schema 1.0 nullability. See https://amazon-ion.github.io/ion-schema/docs/isl-1-0/spec#core-types
```

Comment:
I'm pretty inclined to make modeling ISL 1 a non-goal.

Reply:
I don't think it's an option because of the requirement for ISL 2.0 implementations to also support ISL 1.0